### PR TITLE
Update README to document major concepts in ember-navigator

### PR DESCRIPTION
**Pull request by lukemelia**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,18 +1,110 @@
...
+
+## Concepts
+
+### Router "map"
+
+The router map should seem familiar from ember's router, with some differences. Here's an example of what the router
+map might looks like for an app similar to Twitter:
+
+```js
+tabRouter('tabs', [
```
Comment (knownasilya) [inline README.md:26]: How about doing a full example if the file with imports and all?

Comment (lukemelia) [reply README.md]: Done. Thanks for reviewing. Let me know what else could be clearer or less confusing, and what questions you have after reviewing.

File: README.md [hunk 2]
```diff
@@ -23,47 +23,56 @@ The router map should seem familiar from ember's router, with some differences.
...
-        route('tweet'),
-        // etc...
+import {
+  mount,
+  route,
+  stackRouter,
+  tabRouter
+} from 'ember-navigator';
+
+this.mountedRouter = mount(
```
Comment (knownasilya) [inline README.md:33]: This is in router.js? No export required?

Comment (lukemelia) [reply README.md:33]: Good question. There is not currently a prescriptive or opinionated answer to this question in ember-navigator. I do it in a service in our app. @machty's demo has it in application controller. I'll give some thought to how to document this.

**Comment by machty** [comment]
LGTM :)

**Pull request merged**